client/v2/cli/flag/value.go

```diff
@@ -1,17 +1,15 @@
 package flag
 
 import (
+	"github.com/spf13/pflag"
 	"google.golang.org/protobuf/reflect/protoreflect"
 )
 
-// SimpleValue wraps a simple (non-list and non-map) protobuf value.
-type SimpleValue interface {
-	// Get returns the value.
-	Get() protoreflect.Value
+type Value interface {
+	pflag.Value
+	HasValue
 }
 
-// ListValue wraps a protobuf list/repeating value.
-type ListValue interface {
-	// AppendTo appends the values to the provided list.
-	AppendTo(protoreflect.List)
+type HasValue interface {
+	Get(mutable protoreflect.Value) (protoreflect.Value, error)
```

Comment:
Could you also add godocs to this method and interface? It's not clear to me why we need the `mutable` arg... something to do with List value?

Reply:
added